lib/mayaUsd/ufe/UsdUndoAddNewPrimCommand.cpp

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdUndoAddNewPrimCommand.h"
+
+#include <mayaUsd/ufe/Global.h>
+#include <mayaUsd/ufe/Utils.h>
+
+namespace {
+
+Ufe::Path appendToPath(const Ufe::Path& path, const std::string& name)

```

Comment:
I'm going to leave as-is since this is one of those chunks of code where you occasionally do want to be able to set a breakpoint to stop in a certain situation (dealing with the proxy vs. a prim).